spec/policy/rate_limit/redis_shdict_spec.lua

```diff
@@ -0,0 +1,55 @@
+local env = require('resty.env')
+local ts = require ('apicast.threescale_utils')
+
+local redis_host = env.get('TEST_NGINX_REDIS_HOST') or 'localhost'
+local redis_port = env.get('TEST_NGINX_REDIS_PORT') or 6379
+
+local redis_shdict = require('apicast.policy.rate_limit.redis_shdict')
+
+describe('Redis Shared Dictionary', function()
+    local redis
+    local shdict
+
+    before_each(function()
+        local options = { host = redis_host, port = redis_port, db = 1 }
+        redis = assert(ts.connect_redis(options))
+        shdict = assert(redis_shdict.new(options))
+
+        assert(redis:flushdb())
+    end)
+
+    describe('flush_all', function()
+        it('removes all records', function()
+            assert(redis:set('foo', 'bar'))
+            assert.equal('bar', redis:get('foo'))
+
+            shdict:flush_all()
+
+            assert.equal(ngx.null, redis:get('foo'))
+        end)
+    end)
+
+    describe('incr', function()
+        pending('without default', function()
+            local ret, err = shdict:incr('somekey', 2)
+
+            -- TODO: nginx shdict:incr returns: nil, 'not found' when the key does not exist
+            assert.is_nil(ret)
+            assert.equal('not found', err)
+        end)
+
+        it('has init value', function()
+            local ret = shdict:incr('somekey', 2, 3)
+
+            assert.equal(5, ret)
+            assert.equal('5', redis:get('somekey'))
+        end)
+
+        it('has init_ttl value', function()
+            shdict:incr('somekey', 0, 0, 1)
+
+            assert.equal(1, redis:ttl('somekey'))
```

Comment:
This assertion might fail sometimes. Not sure in practice how often it will, though.

Reply:
Yeah, I was wondering about it. Tried it plenty of times and never got an issue. Maybe with redis running over network it would be a problem.